orangecontrib/bioinformatics/widgets/OWMarkerGenes.py

```diff
@@ -0,0 +1,459 @@
+    icon = 'icons/OWMarkerGenes.svg'
+    priority = 170
+
+    replaces = [
+        'orangecontrib.single_cell.widgets.owmarkergenes.OWMarkerGenes'
+    ]
+
+    class Outputs:
+        genes = widget.Output("Genes", Table)
+
+    want_main_area = True
+
+    selected_group: str = settings.Setting('')
+    selected_db_source: str = settings.Setting('')
+    filter_text: str = settings.Setting('')
+    header_state: bytes = settings.Setting(b'')
+
+    settingsHandler = MarkerGroupContextHandler()
+    selected_genes: List[tuple] = settings.ContextSetting([])
+
+    # class Error(widget.OWWidget.Error):
+    #     file_not_found = widget.Msg("File not found.")
+
+    # class Warning(widget.OWWidget.Warning):
+    #     local_data = widget.Msg("File not found, local cached data is shown.")
+
+    def __init__(self):
+        super().__init__()
+        self._data = None
+        self._available_db_sources = None
+
+        box = gui.widgetBox(self.controlArea, 'Database', margin=0)
+        self.db_source_index = -1
+        self.db_source_cb = gui.comboBox(box, self, 'db_source_index')
+        self.db_source_cb.activated[int].connect(self.handle_source_changed)
+
+        box = gui.widgetBox(self.controlArea, 'Organism', margin=0)
+        self.group_index = -1
+        self.group_cb = gui.comboBox(box, self, 'group_index')
+        self.group_cb.activated[int].connect(self.set_group_index)
+        gui.rubber(self.controlArea)
+
+        # TODO: to avoid this, marker genes table should have 'tax_id' column
+        self.map_group_to_taxid = {'Human': '9606', 'Mouse': '10090'}
+
+        filter_line_edit = gui.lineEdit(
+            self.mainArea, self, "filter_text"
+        )  # type: QLineEdit
+        filter_line_edit.setPlaceholderText("Filter...")
+        filter_line_edit.textEdited.connect(self.set_filter_str)
+
+        self.view = view = QTreeView(
+            rootIsDecorated=False,
+            uniformRowHeights=True,
+            selectionMode=QTreeView.ExtendedSelection,
+            sortingEnabled=True,
+        )
+
+        view.viewport().setMouseTracking(True)
+        self.mainArea.layout().addWidget(view)
+
+        self._load_data()
+        if self.header_state:
+            view.header().restoreState(self.header_state)
+
+    @property
+    def available_db_sources(self) -> dict:
+        return self._available_db_sources
+
+    @available_db_sources.setter
+    def available_db_sources(self, value: dict):
+        self._available_db_sources = value
+
+        items = list(value.keys())
+        try:
+            idx = items.index(self.selected_db_source)
+        except ValueError:
+            idx = -1
+
+        self.db_source_cb.clear()
+        self.db_source_cb.addItems(items)
+
+        if idx != -1:
+            self.db_source_index = idx
+            self.selected_db_source = items[idx]
+        elif items:
+            self.db_source_index = min(max(self.db_source_index, 0), len(items) - 1)
```

Comment:
Can you split all lines that are longer than 80 characters in more lines?

Reply:
The default was 88 :D I've reformated it with line length of 80.